lib/WorseReflection/Core/Inference/Assignments.php

```diff
@@ -13,63 +13,62 @@
 abstract class Assignments implements Countable, IteratorAggregate
 {
     /**
-     * @var array<array-key,array{int,Variable}>
+     * @var array<int, Variable>-
      */
     private array $variables = [];
 
     /**
-     * @param array<array-key, array{int, Variable}> $variables
+     * @param array<int,Variable> $variables
      */
     final public function __construct(array $variables)
     {
-        foreach ($variables as [$offset, $variable]) {
-            $this->variables[] = [ $offset, $variable ];
-        }
+        $this->variables = array_map(function (Variable $v) {
```

Comment:
switched this from `[int, Variable]` to `array<int,Variable>` as variables really shouldn't be validly declared at the same offset.